pkg/k8s/common.go

```diff
@@ -323,39 +324,82 @@ func getPortForwardUrl(config *rest.Config, namespace string, podName string) *u
 	}
 }
 
-func watchForReady(deploymentName *string, readyChan chan<- bool) {
+func watchForReady(deployment *appsv1.Deployment, readyChan chan<- bool) {
 	go func() {
+		lastMsg := ""
+
+		maxUnavailable := deployment.Spec.Strategy.RollingUpdate.MaxUnavailable.IntValue()
+		if maxUnavailable > 0 {
+			log.Warnf("RollingUpdate.MaxUnavailable: %v. This may prevent deployment failures from being detected. Set to 0 to ensure ProgressDeadlineInSeconds is enforced.", maxUnavailable)
+		}
```

Comment:
```suggestion
		if deployment.Spec.Strategy.RollingUpdate != nil &&
			deployment.Spec.Strategy.RollingUpdate.MaxUnavailable != nil &&
			deployment.Spec.Strategy.RollingUpdate.MaxUnavailable.IntValue() > 0 {
			log.Warnf("RollingUpdate.MaxUnavailable: %v. This may prevent deployment failures from being detected. Set to 0 to ensure ProgressDeadlineInSeconds is enforced.", maxUnavailable)
		}
```

`deployment.Spec.Strategy.RollingUpdate` is a pointer and may lead to nil pointer dereference and panic